### DISCUSSION TITLE
Use with Vite?

### DISCUSSION THREAD

**Discussion by martpet** [adobe/react-spectrum#2712] [Q&A]
I'm successfully using it with Vite, however the build creates a large css file, that includes all css from Spectrum, even unused css, not like with Create React App. Do you have any experience with Vite and do you know how to reduce the size of the css file?

**Comment by snowystinger** [comment]
I don't know off the top of my head, I haven't used Vite before. However, could you share a github repo that shows what you're doing? That we can build as well? That may be enough for one of us to be able to help. Not much we can do without that.

**Reply by martpet** [reply]
This example will automatically run the build. Check the size of the CSS in the terminal:
https://stackblitz.com/edit/vitejs-vite-zptqjp?file=src%2FApp.jsx&terminal=build

**Reply by reidbarber** [reply]
Took an initial look but didn't have any luck. My first guess would be adding some custom options to [Dep Optimization Options](https://vitejs.dev/config/#dep-optimization-options) in `vite.config.js`.

**Comment by CanRau** [comment]
Could anyone help/share how to get `@adobe/react-spectrum` working with Vite, please 🥺?

I tried `ssr.noExternal` and other configs but always get something like `Unknown file extension ".css"`

**Reply by reidbarber** [reply]
Here is a codesandbox that uses Vite and has React Spectrum working: https://codesandbox.io/p/sandbox/react-spectrum-vite-86lc7r

**Reply by CanRau** [reply]
interesting, there's nothing special, no config 😅 
Use same react & vite versions, still get the .css error, maybe it's because of streaming ssr or sthg 🤔 
Thanks for sharing anyway 🙏🏼

**Reply by kee-oth** [reply]
@CanRau I'm using react-spectrum with Astro and I'm getting the same error. Were you ever able to come up with a solution? Thanks!

**Reply by CanRau** [reply]
Unfortunately nothing so far sorry 🥺

**Reply by CanRau** [reply]
This works for me https://github.com/adobe/react-spectrum/issues/5401#issuecomment-1815445715
